tls/s2n_cipher_preferences.c

```diff
@@ -1768,4 +1768,20 @@ const struct s2n_cipher_preferences cipher_preferences_20210816_gcm = {
     .suites = cipher_suites_20210816_gcm,
 };
 
+struct s2n_cipher_suite *cipher_suites_rfc9151[] = {
+    /* TLS1.2 */
+    &s2n_ecdhe_ecdsa_with_aes_256_gcm_sha384,
+    &s2n_ecdhe_rsa_with_aes_256_gcm_sha384,
+    &s2n_rsa_with_aes_256_gcm_sha384,
+    &s2n_dhe_rsa_with_aes_256_gcm_sha384,
+
+    /* TLS1.3 */
+    &s2n_tls13_aes_256_gcm_sha384,
+};
+
+const struct s2n_cipher_preferences cipher_preferences_rfc9151 = {
```

Comment:
out of curiosity, is there a reason why we are not using a date version string eg something like 20220822?

Reply:
https://github.com/aws/s2n-tls/blob/main/docs/USAGE-GUIDE.md
> Numbered versions are fixed and will never change.

we try to avoid numbered version where possible since the numbers dont really mean much and since policy cant ever change, it will become outdated at some point.

Reply:
I wouldn't say we avoid them; they're all over. Outside of default, I think all of them are date-versioned. However, in this case I actually think the RFC number is a pretty reasonable version since RFCs are unable to change after being published so we still have immutability and it means a little more than just some date.